doc/smart-switch/pmon/smartswitch-pmon.md

```diff
@@ -0,0 +1,924 @@
+    {
+       "dpuX": {
+                "Ethernet-BPX":  "Ethernet0"
+        },
+    },
+
+]
+```
+On the DPU's platform.json, we can have 
+```
+DPU: {
+
+  // Anything specific to DPU, else remain empty
+
+}
+```
+#### 3.1.4 ModuleBase class API enhancements
+get_base_mac(self):
+```
+    Retrieves the base MAC address for the module
+
+    Returns:
+        A string containing the MAC address in the format 'XX:XX:XX:XX:XX:XX'
+```
+
+get_system_eeprom_info(self):
+```
+    Retrieves the full content of system EEPROM information for the DPU module
+
+    Returns:
+        DPU modules return a dictionary with key value pairs of IDPROM 
+```
+
+get_name(self):
+```
+    Retrieves the name of the device
+
+    Returns:
+        string: The name of the device. Ex; SWITCH, DPU0, DPUX
+```
+
+get_description(self):
+```
+    Retrieves the platform vendor's product description of the module
+
+    Returns:
+        A string, providing the vendor's product description of the module.
+```
+
+get_type(self):
+```
+    Retrieves the type of the module.
+
+    Returns:
+        A string, the module-type from one of the predefined types:
+        MODULE_TYPE_SWITCH, MODULE_TYPE_DPU
+```
+
+get_oper_status(self):
+```
+    Retrieves the operational status of the module
+    This information is not sufficient for debugging complex DPU failures.
+    So, couple of new CLIs will be introduced.
+
+    Returns:
+        A string, the operational status of the module from one of the
+        predefined status values: MODULE_STATUS_EMPTY, MODULE_STATUS_OFFLINE,
+        MODULE_STATUS_FAULT, MODULE_STATUS_PRESENT or MODULE_STATUS_ONLINE
+```
+
+reboot(self, reboot_type):
+```
+    Request to reboot the module
+
+    Args:
+        reboot_type: A string, the type of reboot requested from one of the
+        predefined reboot types: MODULE_REBOOT_DEFAULT,
+        MODULE_REBOOT_CPU_COMPLEX, or MODULE_REBOOT_FPGA_COMPLEX
+
+    Returns:
+        bool: True if the request has been issued successfully, False if not
+```
+
+set_admin_state(self, up):
+```
+    Request to keep the card/DPU in administratively up/down state.
+    Default state is up.
+```
+
+get_maximum_consumed_power(self):
+```
+    Retrieves the maximum power drawn by this module
+
+    Returns:
+        A float, with value of the maximum consumable power of the
+        module.
+```
+
+get_midplane_ip(self):
+```
+    Retrieves the midplane IP-address of the module
+    When called from the DPU, returns the midplane IP-address of the dpu-card.
+    When called from the Switch returns the midplane IP-address of Switch.
+
+    Returns:
+        A string, the IP-address of the module reachable over the midplane
+```
+
+is_midplane_reachable(self):
+```
+    Retrieves the reachability status of the module from the Supervisor or
+    of the Supervisor from the module via the midplane of the modular chassis
+
+    Returns:
+        A bool value, should return True if module is reachable via midplane
+```
```

Comment:
As per our meeting discussion, are we considering implementing an API to retrieve the timeout value configured for errors that occur during the shutdown or boot-up sequence?

Reply:
Not sure if I understood the question right. "is_midplane_reachable" should always return a bool.  For example the ping timeout is a well defined timeout and when that happens the platform will return  False and when it succeeds the platform will return True.